api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineConfigurationManagerTest.java

```diff
@@ -16,182 +16,155 @@
-        Assert.assertFalse(config.getBuckets().isEmpty());
-        Assert.assertFalse(config.getNfsMountOptions().isEmpty());
+    public void shouldGetUnregisteredPipelineConfiguration() {
+        doReturn(TEST_DOCKER_IMAGE).when(pipelineVersionManager).getValidDockerImage(eq(TEST_IMAGE));
+        doReturn(dataStorages).when(dataStorageApiService).getWritableStorages();
 
-        String[] buckets = config.getBuckets().split(";");
+        final PipelineStart vo = getPipelineStartVO();
+        final PipelineConfiguration config = pipelineConfigurationManager.getPipelineConfiguration(vo);
+        assertFalse(config.getBuckets().isEmpty());
+        assertFalse(config.getNfsMountOptions().isEmpty());
 
-        Assert.assertEquals(2, buckets.length);
+        String[] buckets = config.getBuckets().split(";");
+        assertEquals(4, buckets.length);
         for (String bucket : buckets) {
-            Assert.assertTrue(dataStorages.stream().anyMatch(ds -> bucket.equals(ds.getPathMask())));
+            assertTrue(dataStorages.stream().anyMatch(ds -> bucket.equals(ds.getPathMask())));
         }
 
-
-        String[] nfsOptions = config.getNfsMountOptions().split(";");
-
-        Assert.assertEquals(1, nfsOptions.length);
+        String[] nfsOptions = deleteEmptyElements(config.getNfsMountOptions().split(";"));
+        assertEquals(2, nfsOptions.length);
         for (String option : nfsOptions) {
-            if (StringUtils.isNotBlank(option)) {
-                Assert.assertTrue(dataStorages.stream()
-                                      .filter(ds -> ds instanceof NFSDataStorage)
-                                      .anyMatch(ds -> {
-                                          NFSDataStorage nfsDs = (NFSDataStorage) ds;
-                                          return nfsDs.getMountOptions().equals(option) ||
-                                                 option.equals(nfsDs.getMountOptions() + ",ro");
-                                      }));
-            }
+            assertTrue(dataStorages.stream()
+                    .filter(ds -> ds instanceof NFSDataStorage)
+                    .anyMatch(ds -> {
+                        NFSDataStorage nfsDs = (NFSDataStorage) ds;
+                        return nfsDs.getMountOptions().equals(option) ||
+                                option.equals(nfsDs.getMountOptions() + ",ro");
+                    }));
         }
 
-        String[] mountPoints = config.getMountPoints().split(";");
-        for (String mountPoint : mountPoints) {
-            if (StringUtils.isNotBlank(mountPoint)) {
-                Assert.assertTrue(dataStorages.stream().anyMatch(ds -> mountPoint.equals(ds.getMountPoint())));
-            }
-        }
-        //Assert.assertTrue(Arrays.stream(nfsOptions).anyMatch(o -> o.endsWith(",ro")));
+        String[] mountPoints = deleteEmptyElements(config.getMountPoints().split(";"));
+
+        assertEquals(dataStorages.get(0).getMountPoint(), mountPoints[0]);
```

Comment:
Could you please apply the approach from comment above?

Reply:
+